x/snapshot/exported/types.go

```diff
@@ -36,6 +29,39 @@ type ValidatorI interface {
 	IsBonded() bool                        // whether the validator is bonded
 }
 
+// QuadraticWeightFunc returns floor(sqrt(consensusPower)) as the weight
+func QuadraticWeightFunc(consensusPower sdk.Uint) sdk.Uint {
+	bigInt := consensusPower.BigInt()
+
+	return sdk.NewUintFromBigInt(bigInt.Sqrt(bigInt))
+}
+
+// Slasher provides necessary functions to the validator information
+type Slasher interface {
+	IsTombstoned(ctx sdk.Context, consAddr sdk.ConsAddress) bool // whether a validator is tombstoned
+}
+
+// IsTombstoned returns a function that checks if validator is tombstoned
+func IsTombstoned(ctx sdk.Context, slasher Slasher) func(ValidatorI) bool {
+	return func(v ValidatorI) bool {
+		consAdd, err := v.GetConsAddr()
+		if err != nil {
+			return true
+		}
+
+		return slasher.IsTombstoned(ctx, consAdd)
+	}
+}
+
+// IsProxyActive returns a function that checks if validator proxy is active
+func IsProxyActive(ctx sdk.Context, proxyFunc func(sdk.Context, sdk.ValAddress) (sdk.AccAddress, bool)) func(ValidatorI) bool {
```

Comment:
Preference between the two approaches? Can't just reuse `Snapshotter` type from below because of extra methods that are not needed for this function/and not present in reward's Snapshotter.